vnext/ReactWindowsCore/V8JSIRuntimeHolder.cpp

```diff
@@ -16,11 +16,19 @@ class ReactQueueBackedTaskRunner {
  public:
   ReactQueueBackedTaskRunner(std::shared_ptr<facebook::react::MessageQueueThread> jsQueue)
       : jsQueue_(std::move(jsQueue)) {
-    // This thread handles the delayed tasks.
+    // This thread handles the delayed tasks
     std::thread(&ReactQueueBackedTaskRunner::TimerFunc, this).detach();
   }
 
-  ~ReactQueueBackedTaskRunner() {}
+  ~ReactQueueBackedTaskRunner() {
+    stop_requested_ = true;
+
+    delayed_tasks_available_cond_.notify_all();
+
+    // Wait until the thread is done
+    std::unique_lock<std::mutex> worker_stopped_lock(delayed_queue_access_mutex_);
+    delayed_tasks_available_cond_.wait(worker_stopped_lock, [this]() { return timer_stopped_; });
```

Comment:
>this [](start = 61, length = 4)

is it still valid in destructor?

Reply:
maybe we should just pass "timer_stopped_" into lambda

---
In reply to: [459713626](https://github.com/microsoft/react-native-windows/pull/5580#discussion_r459713626) [](ancestors = 459713626)